src/Microsoft.EntityFrameworkCore.AutoHistory/Extensions/DbContextExtensions.cs

```diff
@@ -142,7 +159,11 @@ internal static TAutoHistory AutoHistory<TAutoHistory>(this EntityEntry entry, F
         {
             foreach (var entry in addedEntries)
             {
-                context.Add<TAutoHistory>(entry.AddedHistory(createHistoryFactory));
+                var autoHistory = entry.AutoHistory(createHistoryFactory);
```

Comment:
@isaacgimenez seems like this change (calling .AutoHistory instead of .AddedHistory) broke the EnsureAddedHistory functionality. Is this a mistake or is there a special reason for changing this?

Reply:
Hi @rufer7 seems like a total mistake, There was no intention on changing the behavior. 

I will create a fix to revert it back on the next few days

Reply:
Hi again @rufer7 

PR was done but it has not been merged yet.

Hopefully someone will look into it soon.

Regards.